scripts/release.py

```diff
@@ -441,23 +495,40 @@ def main():
     proceed_text = (
         'WARNING - Running this script will:\n'
         '  - add a tag to %s\n'
+        '  - update version number by pushing a commit to %s\n'
         '  - submit an issue/pull-request to %s\n'
         'If you still wish to proceed type "YES": '
-    ) % (args.repo_url, args.docs_repo_url)
+    ) % (args.repo_url, args.repo_url,  args.docs_repo_url)
 
     proceed = raw_input(proceed_text)
 
     if proceed != 'YES':
         return
 
-    if args.github_token
+    if args.github_token:
         token = args.github_token
     elif os.environ['RPC_GITHUB_TOKEN']:
         token = os.environ['RPC_GITHUB_TOKEN']
     else:
         return
 
-    rpco_repo = Repo(url=args.repo_url, cache_dir=args.cache_dir, bare=True)
+    if args.branch:
+        branch = str(args.branch)
+    else:
+        # Discover branch from current repository
+        try:
+            branch = str(
+                    awk(
+                        git("branch", "--no-color"),
+                        "/*/ { print $2; } ")
+                ).strip()
+        except:
+            raise Exception("Can't discover branch")
```

Comment:
If you change this to
```python
return "The current branch cannot be discovered."
```
The script will fail with a reasonable message instead of an ugly exception failure.